genesis/process/sovereignGenesisBlockCreator.go

```diff
@@ -0,0 +1,286 @@
+package process
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math"
+	"math/big"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/data/block"
+	"github.com/multiversx/mx-chain-core-go/data/transaction"
+	"github.com/multiversx/mx-chain-go/config"
+	"github.com/multiversx/mx-chain-go/factory"
+	"github.com/multiversx/mx-chain-go/genesis"
+	"github.com/multiversx/mx-chain-go/process"
+	"github.com/multiversx/mx-chain-go/update"
+	"github.com/multiversx/mx-chain-go/vm"
+	vmcommon "github.com/multiversx/mx-chain-vm-common-go"
+)
+
+type sovereignGenesisBlockCreator struct {
+	*genesisBlockCreator
+}
+
+// NewSovereignGenesisBlockCreator creates a new sovereign genesis block creator instance
+func NewSovereignGenesisBlockCreator(gbc *genesisBlockCreator) (*sovereignGenesisBlockCreator, error) {
+	if gbc == nil {
+		return nil, errNilGenesisBlockCreator
+	}
+
+	return &sovereignGenesisBlockCreator{
+		genesisBlockCreator: gbc,
+	}, nil
+}
+
+// CreateGenesisBlocks will create sovereign genesis blocks
+func (gbc *sovereignGenesisBlockCreator) CreateGenesisBlocks() (map[uint32]data.HeaderHandler, error) {
+	if !mustDoGenesisProcess(gbc.arg) {
+		return gbc.createSovereignEmptyGenesisBlocks()
+	}
+
+	if mustDoHardForkImportProcess(gbc.arg) {
```

Comment:
you can delete hardfork process. it is deprecated.

Reply:
Deleted it